src/item.cpp

```diff
@@ -9727,11 +9728,11 @@ bool item::process_extinguish( player *carrier, const tripoint &pos )
             break;
     }
     map &here = get_map();
-    if( in_inv && here.has_flag( flag_DEEP_WATER, pos ) ) {
+    if( in_inv && here.has_flag( flag_DEEP_WATER, pos ) && !in_veh) {
```

Comment:
```suggestion
    if( in_inv && !in_veh && here.has_flag( flag_DEEP_WATER, pos ) ) {
```
closing parenthesis should have a space before it, and the quick, pre-calculated value should be checked before the relatively expensive flag check.